backend/internal/controllers/router.go

```diff
@@ -29,7 +30,10 @@ func GetRouter() chi.Router {
 
 	r.Route("/books", func(r chi.Router) {
 		// "localhost:8080/books/{id}
-		r.Get("/{id}", bookController.GetBookDetailsByID)
+		r.Route("/{id}", func(r chi.Router) {
+			//bookController.GetBookDetailsByID)
```

Comment:
Dead code right here.

Also you don't need a `.Route()` call for every slash in the request, this could just be `r.Get("/{id}/{lang}", bookController.GetBookDetailsByID)`. 

You could even flatten this entire section to:

```
r.Get("/books", ...)
r.Get("/books/{id}/{lang}", ...)
```

if you wanted to.